src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+    private static final User SECOND_USER = new User();
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+
+        FIRST_USER.setLogin("BestLogin");
+        FIRST_USER.setPassword(CORRECT_PASSWORD);
+        FIRST_USER.setAge(CORRECT_AGE);
+
+        SECOND_USER.setLogin("GoodLogin");
+        SECOND_USER.setPassword(CORRECT_PASSWORD);
+        SECOND_USER.setAge(CORRECT_AGE);
+    }
+
+    @Test
+    void register_additionUsers_ok() {
+        boolean expect = FIRST_USER.equals(REGISTRATION.register(FIRST_USER));
+        assertTrue(expect);
+        expect = SECOND_USER.equals(REGISTRATION.register(SECOND_USER));
+        assertTrue(expect);
+    }
+
+    @Test
+    void register_checkIdIncrease_ok() {
+        REGISTRATION.register(FIRST_USER);
+        REGISTRATION.register(SECOND_USER);
+        assertEquals(FIRST_USER.getId() + 1, (long) SECOND_USER.getId());
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        FIRST_USER.setLogin(EMPTY);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_loginContainsWhitespaces_notOk() {
+        FIRST_USER.setLogin(INCORRECT_LOGIN_WITH_WHITESPACE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        FIRST_USER.setLogin(null);
+        assertThrows(NullPointerException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_checkPasswordOnDigit_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_DIGIT);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_checkPasswordOnWhitespace_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITH_WHITESPACE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_checkPasswordOnUppercaseLetter_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_UPPERCASE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_checkPasswordLength_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_SHORT);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        FIRST_USER.setPassword(null);
+        assertThrows(NullPointerException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_rangeOfAge_notOk() {
+        FIRST_USER.setAge(INCORRECT_AGE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```

Comment:
```suggestion
    @Test
    void register_lessThanMinAge_notOk() {
        User user = new User();
        user.setLogin("validLogin");
        user.setPassword("validPassword123");
        user.setAge(17); // closest to required, but still invalid
        assertThrows(RuntimeException.class,
                () -> registrationService.register(user));
    }
```
this is what good test can look like
also add test for age equal to required, larger than required, negative age